### PR TITLE
Replace kernel resource strings with real .cu source files

### PR DESCRIPTION
**Pull request by tlemo**
The setup (as implemented in this PR) has 3 key parts:

1. Build system changes to copy the files from under `.../jit/codegen/cuda/runtime` to the build output (and also installing them during a full setup)
2. At runtime, detect the location of the library binary (ex. `torch_cuda.so`). See `torchLibPath()` for details.
3. Pass `--include-path=` to NVRTC to point to the right location

After this, we can simply `#include <nvfuser_runtime/foo.cu>` from the generated code.

### CODE REVIEW

**Comment by jjsjann123** [comment]
I don't see anything wrong with this. Maybe we can have a smoke test PR to upstream to see if it breaks other build environment.

File: torch/csrc/jit/codegen/cuda/executor_utils.cpp [hunk 1]
```diff
@@ -21,18 +28,49 @@ namespace executor_utils {
...
+#else
+  // Used to lookup our library using dladdr()
+  static bool reference_symbol = false;
+
+  Dl_info dl_info = {};
+  TORCH_INTERNAL_ASSERT(dladdr(&reference_symbol, &dl_info));
+  const char* lib_filename = dl_info.dli_fname;
+
+  // TODO: a much better solution would be to use C++17's std::filesystem::path,
+  //  although that's currently not available to us (in the PyTorch codebase)
```
Comment (jjsjann123) [inline torch/csrc/jit/codegen/cuda/executor_utils.cpp:76]: Might be worth to try `<filesystem>` with their CI, C++17 features should be reasonable these days.

**Comment by jjsjann123** [comment]
> I don't see anything wrong with this. Maybe we can have a smoke test PR to upstream to see if it breaks other build environment.

Of course this only makes sense when we add windows support first~

File: torch/csrc/jit/codegen/cuda/executor_utils.cpp (hunk 1 above)
Comment (tlemo) [reply torch/csrc/jit/codegen/cuda/executor_utils.cpp:76]: Unfortunately the build system explicitly passes in std=c++14. This means that even if the toolchain supports newer features they can't be used.

**Pull request closed without merging**